src/Project.php

```diff
@@ -220,13 +220,18 @@ public function uploadObject(string $bucketName, string $objectKey, ?DateTimeImm
     /**
      * @throws UplinkException
      */
-    public function deleteObject(string $bucketName, string $objectKey): ObjectInfo
+    public function deleteObject(string $bucketName, string $objectKey): ?ObjectInfo
     {
         $objectResult = $this->ffi->uplink_delete_object($this->cProject, $bucketName, $objectKey);
         $scope = Scope::exit(fn() => $this->ffi->uplink_free_object_result($objectResult));
 
         Util::throwIfErrorResult($objectResult);
 
+        // In case the object does not exist there is no ObjectInfo but no error either.
```

Comment:
Note, it can also be the case that the object existed, but the user didn't have read permission on it, although it had delete permission.

Reply:
Thanks I changed the comment